src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -62,117 +61,297 @@
-	public PacketNEIRecipe( ByteBuf stream ) throws IOException
+	@Reflected
+	public PacketNEIRecipe()
 	{
-		ByteArrayInputStream bytes = new ByteArrayInputStream( stream.array() );
-		bytes.skip( stream.readerIndex() );
-		NBTTagCompound comp = CompressedStreamTools.readCompressed( bytes );
-		if( comp != null )
+		// automatic.
+	}
+
+	// api
+	public PacketNEIRecipe( NBTTagCompound recipe ) throws IOException
+	{
+		this.recipe = recipe;
+	}
+
+	// @Override
+	// public void serverPacketData( INetworkInfo manager, AppEngPacket packet, EntityPlayer player )
+	// {
+	// EntityPlayerMP pmp = (EntityPlayerMP) player;
+	// Container con = pmp.openContainer;
+	//
+	// if( con instanceof IContainerCraftingPacket )
+	// {
+	// IContainerCraftingPacket cct = (IContainerCraftingPacket) con;
+	// IGridNode node = cct.getNetworkNode();
+	// if( node != null )
+	// {
+	// IGrid grid = node.getGrid();
+	// if( grid == null )
+	// {
+	// return;
+	// }
+	//
+	// IStorageGrid inv = grid.getCache( IStorageGrid.class );
+	// IEnergyGrid energy = grid.getCache( IEnergyGrid.class );
+	// ISecurityGrid security = grid.getCache( ISecurityGrid.class );
+	// IInventory craftMatrix = cct.getInventoryByName( "crafting" );
+	// IInventory playerInventory = cct.getInventoryByName( "player" );
+	//
+	// Actionable realForFake = cct.useRealItems() ? Actionable.MODULATE : Actionable.SIMULATE;
+	//
+	// if( inv != null && this.recipeStack != null && security != null )
+	// {
+	// InventoryCrafting testInv = new InventoryCrafting( new ContainerNull(), 3, 3 );
+	// for( int x = 0; x < 9; x++ )
+	// {
+	// if( this.recipeStack[x] != null && this.recipeStack[x].length > 0 )
+	// {
+	// testInv.setInventorySlotContents( x, this.recipeStack[x][0] );
+	// }
+	// }
+	//
+	// IRecipe r = Platform.findMatchingRecipe( testInv, pmp.worldObj );
+	//
+	// if( r != null && security.hasPermission( player, SecurityPermissions.EXTRACT ) )
+	// {
+	// ItemStack is = r.getCraftingResult( testInv );
+	//
+	// if( is != null )
+	// {
+	// IMEMonitor<IAEItemStack> storage = inv.getItemInventory();
+	// IItemList all = storage.getStorageList();
+	// IPartitionList<IAEItemStack> filter = ItemViewCell.createFilter( cct.getViewCells() );
+	//
+	// for( int x = 0; x < craftMatrix.getSizeInventory(); x++ )
+	// {
+	// ItemStack patternItem = testInv.getStackInSlot( x );
+	//
+	// ItemStack currentItem = craftMatrix.getStackInSlot( x );
+	// if( currentItem != null )
+	// {
+	// testInv.setInventorySlotContents( x, currentItem );
+	// ItemStack newItemStack = r.matches( testInv, pmp.worldObj ) ? r.getCraftingResult( testInv ) : null;
+	// testInv.setInventorySlotContents( x, patternItem );
+	//
+	// if( newItemStack == null || !Platform.isSameItemPrecise( newItemStack, is ) )
+	// {
+	// IAEItemStack in = AEItemStack.create( currentItem );
+	// if( in != null )
+	// {
+	// IAEItemStack out = realForFake == Actionable.SIMULATE ? null : Platform.poweredInsert( energy, storage, in,
+	// cct.getSource() );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)